tests/test_util.py

```diff
@@ -2,11 +2,31 @@
 from illuminatio.util import (
     rand_port,
     add_illuminatio_labels,
+    format_for,
     CLEANUP_LABEL,
     CLEANUP_ON_REQUEST,
+    YAML,
+    JSON,
 )
 
 
+@pytest.mark.parametrize(
+    "filename,format_name,expected",
+    [
+        ("foo.yaml", None, YAML),
+        ("foo.yml", None, YAML),
+        ("foo.json", None, JSON),
+        ("foo", None, JSON),
+        ("foo.yaml", "yaml", YAML),
+        ("foo.yaml", "json", JSON),
+        (None, "yaml", YAML),
+        (None, None, JSON),
+    ],
+)
+def test_format_for(filename, format_name, expected):
+    assert format_for(filename, format_name) == expected
+
+
```

Comment:
It would be nice if we also had tests for `write_formatted` and `read_formatted`